hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/AbstractLogicalConnectionImplementor.java

```diff
@@ -62,10 +62,12 @@ public void afterTransaction() {
 	@Override
 	public void begin() {
 		try {
-			log.trace( "Preparing to begin transaction via JDBC Connection.setAutoCommit(false)" );
-			getConnectionForTransactionManagement().setAutoCommit( false );
+			if ( !isSkipAutoCommitCheck() ) {
```

Comment:
Here is my point about this affecting non-DataSource providers as well.  This boolean-protected block kicks in for `LogicalConnectionManagedImpl` which is used for any `ConnectionProvider` scenarion including JTA DS, non-JTA DS and POCP (plain old connection pool).